Automattic/wp-calypso
# FSE: build Newspacks block independently

#### Changes proposed in this Pull Request

This PR allows to compile the Newspack blocks independently:

* Add files for each block: `blog-posts-block` and `carousel-block`
* Add scripts to be able to compile each block independently
* Enqueue assets files through of the `newspack_blocks_block_args` filter.

#### Testing instructions

The filter is added in this PR, so in order to test you need to apply these changes [in the newspack side](https://github.com/Automattic/newspack-blocks/pull/444).

To compile the posts carousel block:

```
> yarn run newspack-blocks-carousel-block
```

it should create the following files in the `./dist` folder:

```
carousel-block-editor.asset.php
carousel-block-editor.css
carousel-block-editor.js
carousel-block-editor.rtl.css
carousel-block-view.asset.php
carousel-block-view.css
carousel-block-view.js
carousel-block-view.rtl.css
```

Also,...

```
> yarn run newspack-blocks-blog-posts-block
```

```
blog-posts-block-editor.asset.php
blog-posts-block-editor.css
blog-posts-block-editor.js
blog-posts-block-editor.rtl.css
blog-posts-block-view.asset.php
blog-posts-block-view.css
blog-posts-block-view.js
blog-posts-block-view.rtl.css
```

Test that both blocks are rightly added in the editor-canvas.
